src/ext/oblivc/ot.c

```diff
@@ -1033,6 +1080,69 @@ bcipherCryptNoResize(BCipherRandomGen* gen,const char* key,int nonce,
   randomizeBuffer(gen,dest,n);
   memxor(dest,src,n);
 }
+
+
+/* 
+   Data structures and functions for dealing with multithreaded nonce-allocation
+   We use gcc's built-in atomics rather than synchronization primitives. On modern CPUs
+   these translate directly into instructions, so they're pretty quick. The reason for
+   all this complication: setting up OT extensions afresh is really expensive, so we
+   want to be able to reuse a single instance across multiple threads
+   */
+
+typedef struct HonestOTExtNonceTracker
```

Comment:
(optional) Since it's just two pointers, can't we incorporate this directly in `SendMsgArgs` below? You can leave it as is if you think this version is readable, but the extra indirection is kind of bugging me.

Reply:
Actually, an alternate suggestion would be to rename it to have it not just be a nonce tracker, but any future variable that requires global state across threads, split out of the same `{Send,Recv}MsgArgs` object. Not sure what that name would be though. Not global, since we can have other independent objects in flight. `{Send,Recv}MsgArgsShared`? I guess we will just have to rely more on comments than variable names.

Reply:
I kept the indirection so that I could separate the atomics and memory-management logic from the main OT extensions logic. I do see your point though. I think I am more in favor of your second proposal, so barring further comment, that is what I will do, but my opinion is not strong, so if you'd really rather go with the first, I am happy to change it to work that way.